week08/movie/src/components/MovieList.js

```diff
@@ -0,0 +1,60 @@
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+import axios from "axios";
+import Movie from "./Movie";
+
+const MovieLine = styled.div`
+  display: flex;
+  align-items: center;
+  padding: 0px 10px;
+  overflow: auto;
+  scrollbar-width: none;
+`;
+
+class MovieList extends React.Component {
+  state = {
+    isLoading: true,
+    movies: [],
+  };
+  getMovies = async () => {
+    const { url } = this.props;
+    const {
+      data: {
+        data: { movies },
+      },
+    } = await axios.get(`https://yts-proxy.now.sh/list_movies.json${url}`);
+    this.setState({ movies, isLoading: false });
+    console.log(movies);
+  };
+  componentDidMount() {
+    this.getMovies();
+  }
+  render() {
+    const { isLoading, movies } = this.state;
+    return (
+      <div className="container">
+        {isLoading ? (
+          //   <div className="loader">
+          //     <span className="loader_text">Loading...</span>
+          //   </div>
+          console.log(isLoading)
+        ) : (
+          <MovieLine>
+            {movies.map((movie) => (
+              <Movie
+                key={movie.id}
+                id={movie.id}
+                year={movie.year}
+                title={movie.title}
+                summary={movie.summary}
+                poster={movie.medium_cover_image}
+              />
+            ))}
+          </MovieLine>
+        )}
+      </div>
+    );
+  }
+}
```

Comment:
- [x] 3. 함수형 컴포넌트 사용
useState와 useEffect 훅을 사용해 api를 함수형 컴포넌트로 가져오는 예제를 새로 작성해 보았습니다. 클래스형 컴포넌트로 작성해주신 코드와 동일한 기능을 합니다.
```suggestion
const MovieList = ({ url }) => {
  const [isLoading, setIsLoading] = useState(true);
  const [movies, setMovies] = useState([]);

  useEffect(() => {
    const getMovies = async () => {
      try {
        const {
          data: {
            data: { movies },
          },
        } = await axios.get(`https://yts-proxy.now.sh/list_movies.json${url}`);
        setMovies(movies);
        setIsLoading(false);
      } catch (error) {
        console.error(error);
        setIsLoading(false);
      }
    };
    getMovies();
  }, [url]);

  return (
    <div className="container">
      {isLoading ? (
        console.log(isLoading)
      ) : (
        <MovieLine>
          {movies.map((movie) => (
            <Movie
              key={movie.id}
              id={movie.id}
              year={movie.year}
              title={movie.title}
              summary={movie.summary}
              poster={movie.medium_cover_image}
            />
          ))}
        </MovieLine>
      )}
    </div>
  );
};
```